ql/src/java/org/apache/hadoop/hive/ql/exec/vector/VectorAssignRow.java

```diff
@@ -778,7 +778,7 @@ private void assignConvertRowColumn(ColumnVector columnVector, int batchIndex,
               text = new Text();
             }
             text.set(string);
-            ((BytesColumnVector) columnVector).setVal(
+            ((BytesColumnVector) columnVector).setRef(
```

Comment:
Text here is used as a re-usable buffer (if no buffer is passed into the method, a new one is created.  I'm honestly not sure why the String is being Wrapped in a Text, at a quick glance, that seems like it can be removed, however, for the purposes of this PR, this is no immutable and therefore is not a candidate for `setRef`.